test/D2L.Security.OAuth2.IntegrationTests/Keys/Default/Data/JwksProviderTests.cs

```diff
@@ -20,91 +20,171 @@ public class JwksProviderTests {
+			}
 		}
 
 		[Test]
 		public void RequestJwksAsync_HTML_Throws() {
-			IJwksProvider publicKeyProvider = new JwksProvider(
-				new HttpClient(),
-				new Uri( m_host + HTML_PATH )
-			);
-
-			var e = Assert.Throws<PublicKeyLookupFailureException>( () =>
-				publicKeyProvider
-					.RequestJwksAsync()
-					.SafeWait()
+			using( SetupJwkServer( out string host ) )
+			using( HttpClient httpClient = new HttpClient() ) {
+				IJwksProvider publicKeyProvider = new JwksProvider(
+					httpClient,
+					new Uri( host + HTML_PATH )
 				);
 
-			StringAssert.Contains( "<body>", e.Message );
+				var e = Assert.Throws<PublicKeyLookupFailureException>( () =>
+					publicKeyProvider
+						.RequestJwksAsync()
+						.SafeWait()
+					);
+
+				StringAssert.Contains( "<body>", e.Message );
+			}
 		}
 
 		[Test]
 		public void RequestJwksAsync_404_Throws() {
-			IJwksProvider publicKeyProvider = new JwksProvider(
-				new HttpClient(),
-				new Uri( m_host + BAD_PATH )
-			);
+			using( SetupJwkServer( out string host ) )
+			using( HttpClient httpClient = new HttpClient() ) {
+				IJwksProvider publicKeyProvider = new JwksProvider(
+					httpClient,
+					new Uri( host + BAD_PATH )
+				);
 
-			Assert.ThrowsAsync<PublicKeyLookupFailureException>( async () => {
-				JsonWebKeySet jwks = await publicKeyProvider
+				Assert.ThrowsAsync<PublicKeyLookupFailureException>( async () => {
+					JsonWebKeySet jwks = await publicKeyProvider
+						.RequestJwksAsync()
+						.SafeAsync();
+				} );
+			}
+		}
+
+		[Test]
+		public void RequestJwksAsync_CantReachServer_Throws() {
+			using( SetupJwkServer( out string host ) )
+			using( HttpClient httpClient = new HttpClient() ) {
+				IJwksProvider publicKeyProvider = new JwksProvider(
+					httpClient,
+					new Uri( "http://foo.bar.fakesite.isurehopethisisneveravalidTLD" )
+				);
+
+				Assert.ThrowsAsync<PublicKeyLookupFailureException>( async () => {
+					JsonWebKeySet jwks = await publicKeyProvider
 					.RequestJwksAsync()
 					.SafeAsync();
-			} );
+				} );
+			}
 		}
 
 		[Test]
-		public void RequestJwksAsync_CantReachServer_Throws() {
-			IJwksProvider publicKeyProvider = new JwksProvider(
-				new HttpClient(),
-				new Uri( "http://foo.bar.fakesite.isurehopethisisneveravalidTLD" )
-			);
+		public async Task RequestJwkAsync_Success() {
+			using( SetupJwkServer( out string host, hasJwk: true, jwkStatusCode: HttpStatusCode.OK ) )
+			using( HttpClient httpClient = new HttpClient() ) {
+				IJwksProvider jwksProvider = new JwksProvider(
+					httpClient,
+					new Uri( host + GOOD_PATH )
+				);
 
-			Assert.ThrowsAsync<PublicKeyLookupFailureException>( async () => {
-				JsonWebKeySet jwks = await publicKeyProvider
-				.RequestJwksAsync()
-				.SafeAsync();
-			} );
+				JsonWebKeySet jwks = await jwksProvider
+					.RequestJwkAsync( GOOD_JWK_ID )
+					.SafeAsync();
+				Assert.IsNotNull( jwks );
+
+				//jwksServer.AssertWasCalled( x => x.Get( GOOD_JWK_PATH ) );

```

Comment:
nit: remove commented code

Reply:
Yeah, I wanted these calls to work, but my local was having some error loading nunit and was hoping someone smarter would fix it for me- perhaps the library uses a different version for its assertions and we don't have a binding redirect or whatever.